### PR TITLE
New app navigation, components and updated routing

### PR DESCRIPTION
**Pull request by makchamp**
#### This PR introduces new side menu navigation to the app
A side drawer has been added. It can be toggled using a menu button at the top left.
The dashboard, create members and modify members fields are currently clickable.
Everything after the login page is now inside the HomeComponent. The app structure is like so:
HOME COMPONENT => CHILD COMPONENTS { DashboardComponents, ManageTeamsComponent, AuditComponent, etc.. } 

**The default (empty)  route is now the home component.**
As the default route was used for the signin/login page, a new 'login' route was created as a replacement
Child routes of the home component are all selectable side menu options.
The empty route redirects to the dashboard component by default.
This route still cannot be accessed unless authenticated.

**Added a toolbar and made the layout responsive**
I followed the guidlines for the layout from the angular docs:
https://material.angular.io/components/sidenav/overview

### CODE REVIEW

File: client/src/app/app.module.ts [hunk 1]
```diff
@@ -17,7 +16,6 @@ import { CookieService } from 'ngx-cookie-service';
     BrowserModule,
     BrowserAnimationsModule,
     AuthModule,
-    AuthRoutingModule,
```
Comment (RobertBeaudenon) [inline client/src/app/app.module.ts:20]: Why did you delete the routing of auth? keep it even if there is only one route inside we might add other after related to authentication. And we don't want to mix contexts.

File: client/src/app/components/home/home.component.html [hunk 2]
```diff
@@ -1,2 +1,18 @@
-<app-dashboard></app-dashboard>
-<button (click)="logout()">Logout</button>
```
Comment (RobertBeaudenon) [inline client/src/app/components/home/home.component.html:2]: Why did you delete the logout button? Keep it for test purposes until we add it inside the toolbar. Or just create the toolbar using angular material and add it in it.

File: client/src/app/modules/alta-main-routing/alta-main-routing.module.ts [hunk 3]
```diff
@@ -1,18 +1,31 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { AuthTabsComponent } from 'src/app/components/auth-tabs/auth-tabs.component';
+import { DashboardComponent } from 'src/app/components/dashboard/dashboard.component';
 import { HomeComponent } from 'src/app/components/home/home.component';
+import { ManageMembersComponent } from 'src/app/components/manage-members/manage-members.component';
 import { AuthGuard } from '../../guards/auth.guard';
 
 const routes: Routes = [
+  { path: 'login', component: AuthTabsComponent },
```
Comment (RobertBeaudenon) [inline client/src/app/modules/alta-main-routing/alta-main-routing.module.ts:11]: As mentioned above, I think it would be better to recreate the file responsible for routing for authentication stuff. So when you do that please add this route in the other file.

File: client/src/app/modules/alta-main/alta-main.module.ts [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
+  declarations: [
+    HomeComponent,
+    DashboardComponent,
+    SideNavComponent],
+  imports: [
+    CommonModule,
+    MatModule,
+    ReactiveFormsModule,
+    HttpClientModule,
+    RouterModule
```
Comment (RobertBeaudenon) [inline client/src/app/modules/alta-main/alta-main.module.ts:26]: You forgot to add inside your imports the angular material-specific modules that you are using. I would say do not import the whole module library, just do it by a specific module(like I did in the auth module). Just check if its better for performance to just add it by subcomponents that you use.
Screen Shot 2020-10-11 at 7 09 14 PM

File: client/src/app/components/home/home.component.html [hunk 5]
```diff
@@ -1,2 +1,18 @@
...
+<mat-drawer-container class="home-container">
+
+    <mat-drawer #drawer opened="true" mode="side">
+        <app-sidenav></app-sidenav>
+    </mat-drawer>
+
+    <mat-drawer-content class="home-content">
+        <button type="button" mat-button (click)="drawer.toggle()">
```
Comment (RobertBeaudenon) [inline client/src/app/components/home/home.component.html:8]: The button is still not a mat-button by looking at its style in the page.
Screen Shot 2020-10-11 at 7 15 33 PM

File: client/src/app/services/token-interceptor.ts [hunk 6]
```diff
@@ -37,9 +37,10 @@ export class TokenInterceptor implements HttpInterceptor {
     // If token is available in the cookie
     if (token) {
       // We set the Authorization header: token inside the object for every request
-      headersConfig['Authorization'] = `Token ${token}`;
+      const authHeader = 'Authorization';
+      headersConfig[authHeader] = `Token ${token}`;
```
Comment (RobertBeaudenon) [inline client/src/app/services/token-interceptor.ts:41]: you are doing that because the lint is complaining?

File: client/src/app/modules/material/material-module.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  declarations: [],
+  imports: [
+    MatSidenavModule,
+  ],
+  providers: [],
+  exports: [
+    MatSidenavModule,
+  ]
+})
+export class MatModule { }
```
Comment (RobertBeaudenon) [inline client/src/app/modules/material/material-module.ts:14]: What is the reason behind creating this module?

File: client/src/app/modules/alta-main-routing/alta-main-routing.module.ts [hunk 8]
```diff
@@ -1,18 +1,33 @@
...
     canActivate: [AuthGuard], // If user has a valid token he will be able to access comment page
+    children: [
+      {
+        path: '',  // Empty path needed in child too to be able to force a redirect
+        redirectTo: 'dashboard',
+        pathMatch: 'full'
+      },
+      { path: 'dashboard', component: DashboardComponent },
+      { path: 'create-members', component: CreateMembersComponent },
+      { path: 'modify-members', component: ModifyMembersComponent }
```
Comment (RobertBeaudenon) [inline client/src/app/modules/alta-main-routing/alta-main-routing.module.ts:24]: Are you adding a route for the rest of the list in the submenu? Maybe add a TODO on the side.

**Review by RobertBeaudenon — CHANGES_REQUESTED** [review]
I added some comments to the code, to understand why some decisions were made.

File: client/src/app/app.module.ts (hunk 1 above)
Comment (makchamp) [reply client/src/app/app.module.ts:20]: I did not think it was necessary because it was only 1 route and to be able to see the whole picture at once. I can add I back if you think we will add more routes there later.

File: client/src/app/components/home/home.component.html (hunk 2 above)
Comment (makchamp) [reply client/src/app/components/home/home.component.html:2]: yes i was going to add it to the toolbar

File: client/src/app/modules/alta-main/alta-main.module.ts (hunk 4 above)
Comment (makchamp) [reply client/src/app/modules/alta-main/alta-main.module.ts:26]: I wanted to import all the mat modules in one place for simplicity. If the module is big it will effect build time not performance. You think the module will get big enough for it to matter ?

File: client/src/app/components/home/home.component.html (hunk 5 above)
Comment (makchamp) [reply client/src/app/components/home/home.component.html:8]: I think I will just change it to a regular button since it is only temporary anyway.

File: client/src/app/services/token-interceptor.ts (hunk 6 above)
Comment (makchamp) [reply client/src/app/services/token-interceptor.ts:41]: Yes it was complaining locally. After this fix it stopped but now it is complaining from the workflow. I dont know why.

File: client/src/app/modules/alta-main-routing/alta-main-routing.module.ts (hunk 8 above)
Comment (makchamp) [reply client/src/app/modules/alta-main-routing/alta-main-routing.module.ts:24]: No only for modify and create member components for now

File: client/src/app/app.module.ts (hunk 1 above)
Comment (RobertBeaudenon) [reply client/src/app/app.module.ts:20]: Yeah, I would just prefer to keep it in a separate route folder.

File: client/src/app/components/home/home.component.html (hunk 2 above)
Comment (RobertBeaudenon) [reply client/src/app/components/home/home.component.html:2]: perfect.

File: client/src/app/modules/alta-main/alta-main.module.ts (hunk 4 above)
Comment (RobertBeaudenon) [reply client/src/app/modules/alta-main/alta-main.module.ts:28]: So I've looked a bit into it, I think since we are going to use angular material all across the website, we should probably create a specific module that will just be responsible of importing the particular components of angular material that we are using, and then we would be able to import this particular module inside the other modules that need it. So this way it would make the code more organized.
https://stackoverflow.com/questions/57038161/angular-8-is-importing-all-material-modules-into-the-main-app-module-better-th

File: client/src/app/modules/material/material-module.ts (hunk 7 above)
Comment (RobertBeaudenon) [reply client/src/app/modules/material/material-module.ts:20]: So as I mentioned in the conversation about the angular material module, what we can do from now on is add all of our angular material used inside this module, and then if we need those components in other particular modules such as auth-module and alta-main module we would just have to import this MatModule which will be centralizing the import of stuff related to angular material. What do you think about that?

File: client/src/app/services/token-interceptor.ts (hunk 6 above)
Comment (RobertBeaudenon) [reply client/src/app/services/token-interceptor.ts:41]: Perfect 👍

File: client/src/app/components/home/home.component.ts [hunk 9]
```diff
@@ -1,20 +1,46 @@
...
+import { MatSidenav } from '@angular/material/sidenav';
 import { TokenService } from 'src/app/services/token.service';
 import { Router } from '@angular/router';
+import { SidenavService } from 'src/app/services/sidenav.service';
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.css'],
+  providers: [SidenavService],
```
Comment (RobertBeaudenon) [inline client/src/app/components/home/home.component.ts:12]: I would say don't add the SidnavService inside the home component, do it inside the Alta-main-module.

Comment (makchamp) [reply client/src/app/components/home/home.component.ts]: Yes nice catch. This is actually not needed at all. I removed it with 6589be7856d36674cec240568ad3eb202f7d177d

File: client/src/app/modules/material/material-module.ts (hunk 7 above)
Comment (makchamp) [reply client/src/app/modules/material/material-module.ts:36]: Yes exactly. That was my reasoning.

File: client/src/app/modules/alta-main/alta-main.module.ts (hunk 4 above)
Comment (makchamp) [reply client/src/app/modules/alta-main/alta-main.module.ts:28]: Yes exactly. That was my reasoning. The module is indeed MatModule

File: client/src/app/components/home/home.component.ts (hunk 9 above)
Comment (RobertBeaudenon) [reply client/src/app/components/home/home.component.ts]: But you forgot to add it to the alta-main-module inside the providers decorator.

Comment (makchamp) [reply client/src/app/components/home/home.component.ts]: Yes because its not needed there as well

File: client/src/app/services/sidenav.service.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import { MatDrawerToggleResult, MatSidenav } from '@angular/material/sidenav';
+
+@Injectable({
+  providedIn: 'root',
+})
+@Injectable()
+export class SidenavService {
+  private sidenav: MatSidenav;
+
+
```
Comment (RobertBeaudenon) [inline client/src/app/services/sidenav.service.ts:12]: Can you delete those empty lines.

File: client/src/app/services/sidenav.service.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  public setSidenav(sidenav: MatSidenav): void {
+    this.sidenav = sidenav;
+  }
+
+  public open(): Promise<MatDrawerToggleResult> {
+    return this.sidenav.open();
+  }
+
+
```
Comment (RobertBeaudenon) [inline client/src/app/services/sidenav.service.ts:21]: Can you delete this empty line.

File: client/src/app/components/home/home.component.ts [hunk 12]
```diff
@@ -1,20 +1,45 @@
...
+
+  ngOnDestroy(): void {
+    this.mobileQuery.removeEventListener('change', this.mobileQueryListener);
+  }
+
+  ngAfterViewInit(): void {
+    this.sidenavService.setSidenav(this.sidenav);
+  }
 
   logout(): void {
```
Comment (RobertBeaudenon) [inline client/src/app/components/home/home.component.ts:40]: Can you please delete this method since now it's implemented inside toolbar component.

File: client/src/app/components/sidenav/sidenavListing.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+      { title: 'Create Template', routerLink: 'create-template', subMenuOptions: [] },
+    ]
+  },
+  {
+    title: 'Manage Members', routerLink: 'manage-members', subMenuOptions: [
+      { title: 'Create Members', routerLink: 'create-members', subMenuOptions: [] },
+      { title: 'Modify Members', routerLink: 'modify-members', subMenuOptions: [] },
+    ]
+  },
+
```
Comment (RobertBeaudenon) [inline client/src/app/components/sidenav/sidenavListing.ts:19]: Can you please delete this empty line

File: client/src/app/app.module.ts [hunk 14]
```diff
@@ -17,9 +17,9 @@ import { CookieService } from 'ngx-cookie-service';
...
     AltaMainModule,
     AltaMainRoutingModule,
+    AuthRoutingModule,
     HttpClientModule,
```
Comment (RobertBeaudenon) [inline client/src/app/app.module.ts:23]: I think you forgot to add here the module related to angular material that you just created

**Review by RobertBeaudenon — CHANGES_REQUESTED** [review]
I added some little comments.
You forgot to add a file to test the initialization of the side-nav service.
But overall looks good, nice job.

File: client/src/app/components/home/home.component.ts (hunk 12 above)
Comment (makchamp) [reply client/src/app/components/home/home.component.ts:40]: This is actually needed here as the sidenav element is inside the home component.

Comment (RobertBeaudenon) [reply client/src/app/components/home/home.component.ts:40]: No i'm talking about the logout().

Comment (makchamp) [reply client/src/app/components/home/home.component.ts]: yes sorry

**Comment by RobertBeaudenon** [comment]
Thank you for addressing my comments.
Everything looks good, I will merge and we can discuss where we go from here.
